### PR TITLE
fix: Check if chart data in props is valid or not

### LINKED ISSUE #26325: TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
Sentry Issue: [APPSMITH-6SE](https://appsmith.sentry.io/issues/4382323470/?referrer=github_integration)

```
TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
  at forEach (widgets/ChartWidget/component/EChartsDatasetBuilder.ts:26:42)
  at Array.forEach (<anonymous>)
  at <object>.value (widgets/ChartWidget/component/EChartsDatasetBuilder.ts:22:32)
  at _this.getEChartsOptions (widgets/ChartWidget/component/index.tsx:136:34)
  at _this.renderECharts (widgets/ChartWidget/component/index.tsx:195:35)
...
(15 additional frame(s) were not displayed)
```

### LINKED ISSUE #26548: TypeError: a.replace is not a function
Sentry Issue: [APPSMITH-6W3](https://appsmith.sentry.io/issues/4408929755/?referrer=github_integration)

```
TypeError: a.replace is not a function
  at m (../node_modules/fusioncharts/fusioncharts.js:13:176102)
  at a.draw (../node_modules/fusioncharts/fusioncharts.widgets.js:1:64053)
  at t.syncDraw (../node_modules/fusioncharts/fusioncharts.js:13:202559)
  at s (../node_modules/fusioncharts/fusioncharts.js:13:198841)
  at f (../node_modules/fusioncharts/fusioncharts.js:13:199127)
...
(7 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by rajatagrawal**
Fixes #26325
Fixes #26548

### CODE REVIEW

File: app/client/src/widgets/ChartWidget/widget/index.tsx [hunk 1]
```diff
@@ -17,11 +17,33 @@ import type { AutocompletionDefinitions } from "widgets/constants";
...
+export const isChartDataValid = (props: ChartWidgetProps): boolean => {
+  switch (props.chartType) {
+    case "CUSTOM_FUSION_CHART": {
+      return !!props.customFusionChartConfig;
+    }
+    default: {
+      for (const seriesID in props.chartData) {
+        const seriesData = props.chartData[seriesID];
+
+        if (!seriesData) {
```
Comment (sbalaji1192) [inline app/client/src/widgets/ChartWidget/widget/index.tsx:35]: combine these two if condition

**Comment by sbalaji1192** [comment]
Why do we need to perform this check on the widget, Can we do this in the validation itself?

**Comment by rajatagrawal** [comment]
> Why do we need to perform this check on the widget, Can we do this in the validation itself?

@sbalaji1192 

We have a check in the validation. But I don't why we are still getting undefined data. Last time this issue was fixed, I added checks in the validation only. The issue has regressed even then.

File: app/client/src/widgets/ChartWidget/widget/index.tsx (hunk 1 above)
Comment (rajatagrawal) [reply app/client/src/widgets/ChartWidget/widget/index.tsx]: Done

**Comment by sbalaji1192** [comment]
@rajatagrawal Let's figure out the root cause and fix this issue on the validation itself. From a widget perspective, it should always receive valid values for the properties defined in the property pane. We can ensure this by writing proper validation functions.
Ideally, the widget should not be firing sentry errors. We need to modify log.error to report the error to sentry as part of diff PR.

**Pull request closed without merging**